pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -3766,13 +3773,26 @@ func (ss *SemaphoreStatus) LockWaiting(holderKey, lockKey string, currentHolders
 	return true
 }
 
+func GetHolderKeyVersion() HoldingNameVersion {
+	switch os.Getenv("HOLDER_KEY_VERSION") {
+	case "v2":
+		return HoldingNameV2
+	default:
+		return HoldingNameV1
+	}
+}
+func GetHoldingNameV2OrV1(holderKey string) string {
```

Comment:
```suggestion
func GetHoldingName(holderKey string) string {
```

We can simplify the function name, no?

Reply:
+1